tests/python/pants_test/base/test_hash_utils.py

```diff
@@ -31,6 +33,65 @@ def test_hash_file(self):
 
       self.assertEqual(expected_hash.hexdigest(), hash_file(fd.name, digest=hashlib.md5()))
 
+  def test_hash_dir_invalid(self):
+    with temporary_file_path() as path:
+      with self.assertRaises(TypeError):
+        hash_dir(path)
+      with self.assertRaises(ValueError):
+        hash_dir(Path(path))
+
+  def test_hash_dir(self):
+    with temporary_dir() as root1:
+      root1_path = Path(root1)
+      with root1_path.joinpath('a').open(mode='wb') as fd:
+        fd.write(b'jake jones')
+      with root1_path.joinpath('b').open(mode='wb') as fd:
+        fd.write(b'jane george')
+      hash1 = hash_dir(root1_path)
+
+    with temporary_dir() as root2:
+      root2_path = Path(root2)
+      with root2_path.joinpath('a').open(mode='wb') as fd:
```

Comment:
This can be replaced with one line: `my_path.write_text(“foo”)`

This, and `read_text()` are my favorite things about pathlib.Path. You rarely have to use open() as a result

Reply:
Nice! Fixed.